webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/services/DisplayNode.scala

```diff
@@ -95,7 +96,11 @@ object DisplayNode extends Loggable {
   private def loadComplianceBar(nodeInfo: Option[NodeInfo]): Option[JsArray] = {
     for {
       node            <- nodeInfo
-      nodeCompliance  <- asyncComplianceService.nodeCompliance(node.id)
+      // Get only user rules.
+      rules           <- roRuleRepository.getIds(false).toBox.toOption
+      // We don't want to continue if we have no user rules, and an empty set will put all rules, and here systems rule only
+      if rules.size > 0
```

Comment:
this is a feature of for comprehension that we don't use anywhere else, and is not compatible with most other for comprehension (it won't work with zio for ex, because there's no `withFilter`).